### ISSUE TITLE
Bones - Woocommerce

### ISSUE THREAD

**Issue by BonesUser** [eddiemachado/bones#753]
Hello,

I'm having trouble integrating Woocommerce into the Bones theme, which works beautifully by the way.

Situation can be found at: http://www.kaffee-aus-holland.de/home/ 

As you can see, the page called "shop" does not accept the layout applied to the other pages. I tried the following, found at http://docs.woothemes.com/document/third-party-custom-theme-compatibility/:

1. Adding woocommerce.php to the theme, replacing the_loop with <?php woocommerce_content(); ?>

This alligns the product in the shop to the left, but doesn't apply layout.

2. Using hooks in functions.php:

remove_action( 'woocommerce_before_main_content', 'woocommerce_output_content_wrapper', 10);
remove_action( 'woocommerce_after_main_content', 'woocommerce_output_content_wrapper_end', 10);
add_action('woocommerce_before_main_content', 'instyle_wrapper_start', 10);
add_action('woocommerce_after_main_content', 'instyle_wrapper_end', 10);
function instyle_wrapper_start() {
  echo '<div id="main" class="hentry cf">';
}
function instyle_wrapper_end() {
  echo '</div>';
}

Both options, even after lots and lots of tweaks and try-outs did not work out. Please let me know if you have any idea on how to approach this problem.

All the best!

**Comment by Hlsgs** [comment]
Hey!
My preferred method is no 1, for variuos reasons witch I will not go through here.
Have you also added to functions.php:
add_action( 'after_setup_theme', 'woocommerce_support' );
function woocommerce_support() {
add_theme_support( 'woocommerce' );
} 
?

If you want clean start, I also recommend:
add_filter( 'woocommerce_enqueue_styles', 'xxx_dequeue_styles' );
function xxx_dequeue_styles( $enqueue_styles ) {
unset( $enqueue_styles['woocommerce-general'] ); // Remove the gloss
unset( $enqueue_styles['woocommerce-layout'] ); // Remove the layout
unset( $enqueue_styles['woocommerce-smallscreen'] ); // Remove the smallscreen optimisation
return $enqueue_styles;
} 

to prevent the loading of woocommerce styles.

Cheers!

**Comment by Hlsgs** [comment]
Ah, wait, i think you forgot to add <div id="inner-content" class="wrap... to your woocommerce.php
Use page.php as a start for your woocommerce.php file.

**Comment by BonesUser** [comment]
Let me check, just a sec!

**Comment by BonesUser** [comment]
I added <div id="inner-content" class="wrap... to my woocommerce.php file, so that's not it. I did add the woocommerce support tag to the functions.php, but this doesn't change it :( 

Do you have a minute for a small skype conversation? I'm running in circles to be honest.

**Comment by Hlsgs** [comment]
Heres an example of a propperly working woocommerce.php:

<?php
/*
 Template Name: Woocommerce Template
*/
?>

<?php get_header(); ?>

			<div id="content">

				<div id="inner-content" class="wrap cf">

						<div id="main" class="cf" role="main" itemscope itemprop="mainContentOfPage">

						<?php wc_print_notices(); ?>

						<?php do_action( 'breadcrumb'); ?>

							<?php woocommerce_content(); ?>

						</div>

						<?php get_sidebar(); ?>

				</div>

			</div>

<?php get_footer(); ?>

**Comment by Hlsgs** [comment]
replace what you have and it will work. you can always undo if its not right. sorry, dun use skype.

**Comment by BonesUser** [comment]
Thanks for the woocommerce.php! I replaced the Woocommerce.php with your template. Does allign the product, but the layout from the other pages (which is currently the yellow box) doesn't aplly yet..

**Comment by Hlsgs** [comment]
i see that you are fine now. just put the yellow background on #main not article.
cheers!

**Comment by BonesUser** [comment]
Uhm, this is the point where I don't follow. You mean in _base.css? Adding the specifications of entry-content to #main?

**Comment by Hlsgs** [comment]
yes. and add in woocommerce.php the classes that #main has in page.php(<main id="main" class="m-all t-2of3 d-5of7 cf" role="main" itemscope="" itemprop="mainContentOfPage" itemtype="http://schema.org/Blog">)

the code i gave you is custom as i don't use the grid included with bones.

**Comment by BonesUser** [comment]
It works. 

You, sir, thank you very very, very much :)

**Comment by BonesUser** [comment]
..and now I have to find a way to retrieve the widget position on the right

**Comment by Hlsgs** [comment]
No problem. but please take my advice as you may complicate yourself further.
Start from scratch. Copy the code from page.php in woocommerce.php and start removing what you dont need + replace the loop with <?php woocommerce_content(); ?>.
That way you will have consistency across all your pages and it will be easier to style them.(now you're missing article section etc, witch is why you have no padding.
this will save you ALOT of trouble.
Cheers!

**Comment by Hlsgs** [comment]
if you do this, all will just fall into place :)

**Comment by Hlsgs** [comment]
happy developing with our beloved bones :)

**Comment by BonesUser** [comment]
I'll give it a shot, thanks again! I'll post this tread when I get results.

**Comment by Hlsgs** [comment]
ill help you out if im aroud if you dont manage, but its best that you master this yourself.

**Comment by BonesUser** [comment]
Of course, that's what I want. I need the whole Woocommerce skillset :)

**Comment by Hlsgs** [comment]
hey, leave article and section in place if you want consistency. just remove coomments and byline. 
i fyou need different styles for the above for the shop page, Woo adds the 'woocommerce' class o the body of its pages(product archive, single product etc. these that use woocommerce.php as a template ;)

**Comment by BonesUser** [comment]
I removed comments and byline, they're not needed at Woocommerce pages indeed. should I remove the loop as well? And I don't fully understand what section and article have to do with my woocommerce.php layout.

**Comment by Hlsgs** [comment]
they dont. but it helps with styling to have the same content structure throughout you entire site, including styles that come with bones(for example, on your site, the left and right padding on section.entry-content). so, beside comments and byline, remove:
- everything inside <section>, and put <?php woocommerce_content(); ?> there
- all the php between and including <?php endwhile; else : ?> and <?php endif; ?>

**Comment by BonesUser** [comment]
If I remove the whole loop, I don't get the styling which applies to the section, agreed. So, I cleared the section tag and added <?php woocommerce_content(); ?> there. 

Saving and replacing leaves me with an endless loop I suppose, because the loading takes a lot of time. 

If I proceed and remove  the <?php endwhile; else : ?> and <?php endif; ?> tags, including the article code which is in it, I get an error at the closing line of the file. 

I'm realy close now, I'm just overlooking something minor.

**Comment by BonesUser** [comment]
This is what I end up with, after adjusting:

<?php get_header(); ?>

			<div id="content">
				<div id="inner-content" class="wrap cf">

						<main id="main" class="m-all t-2of3 d-5of7 cf" role="main" itemscope itemprop="mainContentOfPage" itemtype="http://schema.org/Blog">

							<?php if (have_posts()) : while (have_posts()) : the_post(); ?>

							<article id="post-<?php the_ID(); ?>" <?php post_class( 'cf' ); ?> role="article" itemscope itemtype="http://schema.org/BlogPosting">

								<header class="article-header">

									<h1 class="page-title" itemprop="headline"><?php the_title(); ?></h1>

								</header> <?php // end article header ?>

								<section class="entry-content cf" itemprop="articleBody">
									<?php woocommerce_content(); ?>
								</section> <?php // end article section ?>

								<footer class="article-footer cf">

								</footer>

							</article>


						</main>

						<?php get_sidebar(); ?>

				</div>

			</div>

<?php get_footer(); ?>

**Comment by Hlsgs** [comment]
yeah. you need to also remove <?php if (have_posts()) : while (have_posts()) : the_post(); ?>
that is part of the loop, sorry, i thought that was obvious cuz i told you to remove endif and endwhile.
so, duh! :P

**Comment by Hlsgs** [comment]
remove the header also, cuz the_tiltle() only works in the loop :)

**Comment by Hlsgs** [comment]
if you get further errors also remove id="post-<?php the_ID(); ?>" <?php post_class( 'cf' ); ?> from article and replace with class="cf"

**Comment by BonesUser** [comment]
It works. Thanks again for your time. In retrospect it's always obvious, but I get it now! The next step for me now is to get a category overview, when hitting the shop button. Do you happen to know the quickest, most effective way?

**Comment by Hlsgs** [comment]
glad you got it :)
what do you mean by category overview?

**Comment by BonesUser** [comment]
Hi Hlsgs,

Before I want to show different categories, I ran into another small problem.

At _768up I would like to show two Woocommerce product per row, instead of three. Do you know where and how I can customize this?

Thanks in advance for your reply!

**Comment by Hlsgs** [comment]
hey again.
i would recommend that you remove woocommerce's default styles and style your website from scratch.
`
add_filter( 'woocommerce_enqueue_styles', 'xxx_dequeue_styles' );
function xxx_dequeue_styles( $enqueue_styles ) {
unset( $enqueue_styles['woocommerce-general'] ); // Remove the gloss
unset( $enqueue_styles['woocommerce-layout'] ); // Remove the layout
unset( $enqueue_styles['woocommerce-smallscreen'] ); // Remove the smallscreen optimisation
return $enqueue_styles;
} 
`

after that you are free to either use the grid that comes with bones by adding classes to the html or extending with @extend if you use sass or just manually set the columns

**Comment by BonesUser** [comment]
I'm not sure how to implement this. Where do I remove the default styles with the piece of code you provided? And how do I implement the grid that comes with bones?

Isn't just setting the amount of products for a specific _up css file an option? I'm not exactly sure how to reason on this as I already did a fair amount of styling on the products etcetera.

Again, thanks for helping me out. It is much appreciated!

**Comment by Hlsgs** [comment]
put that code in functions.php then see how the website looks.

alternatively, keep the styles if you need them and negate what doesn't suit you in the respective breakpoints. for example add to 480up(if you want the distance between items to be 60px and two products per row):
ul.products {
    li.products {
        margin-right: 0px !important;
        float: left !important;
        width: 100% !important;
        &:nth-child(2n) {
           padding-left: 30px; // half the 100px gutter
        }   
        &:nth-child(2n+1) {
           padding-right: 30px; // half the 100px gutter
        }   
}



but i honestly encourage you to uset woocommerces styles completly and style everything yourself.

**Comment by BonesUser** [comment]
I agree on styling everything from scratch, this will give me more insight on how to approach this situation in the future.

So the question remains: how do I setup my own styling for the Woocommerce product pages? Please let me know! And thanks again.

**Comment by dirad** [comment]
hi dude, 
First - the recommended way to eliminate all woo styling is ```add_filter( 'woocommerce_enqueue_styles', '__return_empty_array' );```

Second, for styling on your own, take a look at the generated HTML on the product pages, and write css according to the classes. You can make your own classes for the containing div/elements, and style them as well. Lots of good dos here > http://docs.woothemes.com/documentation/plugins/woocommerce/woocommerce-codex/

good luck!

**Comment by Hlsgs** [comment]
hey ppl.
both methods are equivalent, with the unset one having the advantage of being able to only remove say styling, yet keep responsiveness.

@BonesUser one way to go about it is to take the scss files from wp-content\plugins\woocommerce\assets\css copy them to your theme and import them into your style.scss(maybe only teh ones you need/want). then you can modify them and in the process learn a few things about woo. cheers!
